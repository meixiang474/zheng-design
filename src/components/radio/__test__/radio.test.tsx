import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Radio } from '../index';
import { color } from '../../shared/styles';

const testfn = jest.fn();
const disablefn = jest.fn();

describe('test Radio component', () => {
  it('should checked when clicked', () => {
    const wrapper = render(<Radio label="test" onChange={testfn} />);
    expect(wrapper).toMatchSnapshot();
    const input = wrapper.container.querySelector('input');
    expect(testfn).not.toHaveBeenCalled();
    fireEvent.click(input);
    expect(testfn).toHaveBeenCalled();
  });
  it('should render extra text', () => {
    const wrapper = render(
      <Radio
        label="test"
        error="errortext"
        description="description"
        onChange={testfn}
      />
    );
    expect(wrapper).toMatchSnapshot();
    const errortext = wrapper.getByText('errortext');
    expect(errortext).toHaveStyle(`color: ${color.negative}`);
    const description = wrapper.getByText('description');
    expect(description).toHaveStyle(`color: ${color.mediumdark}`);
  });
  it('should hide label', () => {
    const wrapper = render(<Radio label="test" hideLabel />);
    expect(wrapper).toMatchSnapshot();
    const text = wrapper.getByText('test');
    expect(text).toHaveStyle('clip-path: inset(100%)');
  });
  it('should render correct with disabled', () => {
    const wrapper = render(
      <Radio label="test" disabled onChange={disablefn} />
    );
    expect(wrapper).toMatchSnapshot();
    const text = wrapper.getByText('test');
    fireEvent.click(text);
    expect(disablefn).not.toHaveBeenCalled();
  });
});
