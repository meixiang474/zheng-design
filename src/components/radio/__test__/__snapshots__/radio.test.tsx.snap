// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Radio component should checked when clicked 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKgILt fMWSsO"
      >
        <label
          class="sc-bdfBwQ cUVqiM"
        >
          <input
            class="sc-eCssSg efwVFi"
            type="radio"
          />
          <span>
            <span
              class="sc-gsTCUz ivAixf"
            >
              test
            </span>
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKgILt fMWSsO"
    >
      <label
        class="sc-bdfBwQ cUVqiM"
      >
        <input
          class="sc-eCssSg efwVFi"
          type="radio"
        />
        <span>
          <span
            class="sc-gsTCUz ivAixf"
          >
            test
          </span>
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Radio component should hide label 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKgILt fMWSsO"
      >
        <label
          class="sc-bdfBwQ cUVqiM"
        >
          <input
            class="sc-eCssSg efwVFi"
            type="radio"
          />
          <span>
            <span
              class="sc-gsTCUz NpwnP"
            >
              test
            </span>
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKgILt fMWSsO"
    >
      <label
        class="sc-bdfBwQ cUVqiM"
      >
        <input
          class="sc-eCssSg efwVFi"
          type="radio"
        />
        <span>
          <span
            class="sc-gsTCUz NpwnP"
          >
            test
          </span>
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Radio component should render correct with disabled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKgILt fMWSsO"
      >
        <label
          class="sc-bdfBwQ kmNSUV"
          disabled=""
        >
          <input
            class="sc-eCssSg efwVFi"
            disabled=""
            type="radio"
          />
          <span>
            <span
              class="sc-gsTCUz ivAixf"
            >
              test
            </span>
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKgILt fMWSsO"
    >
      <label
        class="sc-bdfBwQ kmNSUV"
        disabled=""
      >
        <input
          class="sc-eCssSg efwVFi"
          disabled=""
          type="radio"
        />
        <span>
          <span
            class="sc-gsTCUz ivAixf"
          >
            test
          </span>
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Radio component should render extra text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKgILt fMWSsO"
      >
        <label
          class="sc-bdfBwQ cUVqiM"
        >
          <input
            class="sc-eCssSg efwVFi"
            type="radio"
          />
          <span>
            <span
              class="sc-gsTCUz ivAixf"
            >
              test
            </span>
          </span>
        </label>
        <span
          class="sc-jSgupP dkZCYz"
        >
          errortext
        </span>
        <div
          class="sc-dlfnbm ifrNwz"
        >
          description
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKgILt fMWSsO"
    >
      <label
        class="sc-bdfBwQ cUVqiM"
      >
        <input
          class="sc-eCssSg efwVFi"
          type="radio"
        />
        <span>
          <span
            class="sc-gsTCUz ivAixf"
          >
            test
          </span>
        </span>
      </label>
      <span
        class="sc-jSgupP dkZCYz"
      >
        errortext
      </span>
      <div
        class="sc-dlfnbm ifrNwz"
      >
        description
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
